### PR TITLE
Release

### PR DESCRIPTION
**Pull request by nilsgstrabo**

### CODE REVIEW

File: pkg/apis/deployment/radixcomponent.go [hunk 1]
```diff
@@ -100,13 +104,56 @@ func getRadixComponentsForEnv(radixApplication *v1.RadixApplication, env string,
...
 	return components
 }
 
+func GetAuthenticationForComponent(componentAuthentication *v1.Authentication, environmentAuthentication *v1.Authentication) *v1.Authentication {
+	var authentication *v1.Authentication
+
+	if componentAuthentication == nil && environmentAuthentication == nil {
+		authentication = nil
+	} else if componentAuthentication == nil {
+		authentication = environmentAuthentication.DeepCopy()
```
Comment (satr) [inline pkg/apis/deployment/radixcomponent.go:122]: I might be wrong, but I thought that environment value always overrides component value. If I understood logic correct.

File: pkg/apis/deployment/radixcomponent.go [hunk 2]
```diff
@@ -100,13 +104,56 @@ func getRadixComponentsForEnv(radixApplication *v1.RadixApplication, env string,
...
+
+	return authentication
+}
+
+func GetClientCertificateForComponent(componentCertificate *v1.ClientCertificate, environmentCertificate *v1.ClientCertificate) *v1.ClientCertificate {
+	var certificate *v1.ClientCertificate
+	if componentCertificate == nil && environmentCertificate == nil {
+		certificate = nil
+	} else if componentCertificate == nil {
+		certificate = environmentCertificate.DeepCopy()
```
Comment (satr) [inline pkg/apis/deployment/radixcomponent.go:139]: Same question as above

File: pkg/apis/deployment/secrets.go [hunk 3]
```diff
@@ -102,6 +102,18 @@ func (deploy *Deployment) createOrUpdateSecretsForComponent(registration *radixv
 		}
 	}
 
+	clientCertificateSecretName := utils.GetComponentClientCertificateSecretName(component.GetName())
```
Comment (satr) [inline pkg/apis/deployment/secrets.go:105]: `secrtificateExists := deploy.kubeutil.SecretExists(ns, clientCertificateSecretName)` can be extracted here - to reduce repeatable code

File: pkg/apis/deployment/secrets.go [hunk 4]
```diff
@@ -311,3 +346,14 @@ func (deploy *Deployment) removeOrphanedSecrets(ns, app, component, secretName s
...
+func IsSecretRequiredForClientCertificate(clientCertificate *radixv1.ClientCertificate) bool {
+	if clientCertificate != nil {
+		certificateConfig := parseClientCertificateConfiguration(*clientCertificate)
+		if *certificateConfig.PassCertificateToUpstream || *certificateConfig.Verification != radixv1.VerificationTypeOff {
+			return true
+		}
+	}
+
+	return false
+}
```
Comment (satr) [inline pkg/apis/deployment/secrets.go:359]: Suggested change
```
	if clientCertificate == nil {
               return false
        }
        certificateConfig := parseClientCertificateConfiguration(*clientCertificate)
	return *certificateConfig.PassCertificateToUpstream || *certificateConfig.Verification != radixv1.VerificationTypeOff
```

File: pkg/apis/radixvalidators/validate_ra.go [hunk 5]
```diff
@@ -503,6 +518,42 @@ func validateJobComponents(app *radixv1.RadixApplication) []error {
...
+		string(v1.VerificationTypeOff),
+		string(v1.VerificationTypeOn),
+		string(v1.VerificationTypeOptional),
+		string(v1.VerificationTypeOptionalNoCa),
+	}
+
+	actualValue := string(*verificationType)
+	if !slice.ContainsString(validValues, actualValue) {
+		return InvalidVerificationType(actualValue)
+	} else {
```
Comment (satr) [inline pkg/apis/radixvalidators/validate_ra.go:552]: Not needed `else`

**Review by satr — APPROVED** [review]
Just minor suggestions, can be ignored

File: pkg/apis/deployment/radixcomponent.go (hunk 1 above)
Comment (nilsgstrabo) [reply pkg/apis/deployment/radixcomponent.go:122]: It uses environmentConfig is componentConfig is nil

File: pkg/apis/deployment/radixcomponent.go (hunk 2 above)
Comment (nilsgstrabo) [reply pkg/apis/deployment/radixcomponent.go:139]: Same here

File: pkg/apis/deployment/secrets.go (hunk 3 above)
Comment (nilsgstrabo) [reply pkg/apis/deployment/secrets.go:105]: Not sure I understand

File: pkg/apis/deployment/secrets.go (hunk 4 above)
Comment (nilsgstrabo) [reply pkg/apis/deployment/secrets.go:359]: Agree. I'll send a message to Fredrik

**Pull request merged**